mitiq/zne/inference.py

```diff
@@ -417,6 +435,37 @@ class PolyFactory(BatchedFactory):
         RichardsonFactory and LinearFactory are special cases of PolyFactory.
     """
 
+    @staticmethod
+    def extrapolate(
+        scale_factors: List[float], exp_values: List[float], order: int,
+    ) -> Tuple[float, List[float]]:
+        """Static method which evaluates a polynomial extrapolation to the
+        zero-noise limit.
+
+        Args:
+            scale_factors: The array of noise scale factors.
+            exp_values: The array of expectation values.
+            order: The extrapolation order (degree of the polynomial fit).
+                It cannot exceed len(scale_factors) - 1.
+
+        Returns:
+            (zero_lim, opt_params): Where "zero_lim" is the zero-noise limit
+            and "opt_params" is the parameter array of the best fitting model.
+
+        Raises:
+            ExtrapolationWarning: If the extrapolation fit is ill-conditioned.
+
+        Note:
+            This method computes the zero-noise limit only from the information
+            contained in the input arguments. To extrapolate from the internal
+            data of an instantiated Factory object, the bound method
+            ".reduce()" should be called instead.
+        """
+
+        opt_params = mitiq_polyfit(scale_factors, exp_values, order)
+        zero_lim = opt_params[-1]
+        return zero_lim, opt_params
```

Comment:
Suggestion: Include a `full_output` such that if `full_output == True`, return both `zero_lim` and `opt_params`, else return only `zero_lim`.

I'm fine either way.